backend/src/main/java/ca/mcgill/purposeful/controller/AppUserController.java

```diff
@@ -81,4 +90,123 @@ public ResponseEntity<AppUserDto> registerModerator(@RequestBody AppUserDto appU
     return ResponseEntity.status(HttpStatus.CREATED).body(registeredUser);
   }
 
+  /**
+   * PUT method to update the names of a regular user
+   * @param appUserDto - the user to modify the names
+   * @return the modified user
+   * 
+   * @author Enzo Benoit-Jeannin
+   */
+    @PutMapping(value = {"/regular",
+      "/regular/"}, consumes = "application/json", produces = "application/json")
+    @PreAuthorize("hasAnyAuthority('Owner', 'Moderator', 'User')")
+    public ResponseEntity<AppUserDto> updateRegularUser(@RequestBody AppUserDto appUserDto) {
+      // Unpack the DTO
+      if (appUserDto == null) {
+        throw new GlobalException(HttpStatus.BAD_REQUEST, "AppUserDto is null");
+      }
+      String email = appUserDto.getEmail();
+      String firstname = appUserDto.getFirstname();
+      String lastname = appUserDto.getLastname();
+  
+      // Register the user
+      AppUserDto registeredUser = DtoUtility.convertToDto(
+          appUserService.modifyUserNames(email, firstname, lastname));
+  
+      return ResponseEntity.status(HttpStatus.OK).body(registeredUser);
+    }
+
+    /**
+     * PUT method to update the password of a regular user
+     * @param appUserDto - the user to modify the password
+     * @return the modified user
+     * 
+     * @author Enzo Benoit-Jeannin
+     */
+    @PutMapping(value = {"/regular/password",
+      "/regular/password/"}, consumes = "application/json", produces = "application/json")
+    @PreAuthorize("hasAnyAuthority('Owner', 'Moderator', 'User')")
```

Comment:
Same comment here :
This endpoint should be restricted to the concerned user only so that other users cannot modify the information. Is that authentication done?